core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricTokenServiceClient.scala

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.common.CommonUtils
+import com.microsoft.azure.synapse.ml.logging.Usage.FabricConstants._
+import java.lang.management.ManagementFactory
+import java.net.URL
+import java.net.InetAddress
+import java.util.UUID
+import spray.json.{JsArray, JsObject, JsValue, _}
+
+class FabricTokenServiceClient {
+  private val resourceMapping = Map(
+    "https://storage.azure.com" -> "storage",
+    "storage" -> "storage",
+    "https://analysis.windows.net/powerbi/api" -> "pbi",
+    "pbi" -> "pbi",
+    "https://vault.azure.net" -> "keyvault",
+    "keyvault" -> "keyvault",
+    "https://kusto.kusto.windows.net" -> "kusto",
+    "kusto" -> "kusto"
+  )
+
+  private val hostname = InetAddress.getLocalHost.getHostName
+  private val processDetail = ManagementFactory.getRuntimeMXBean.getName
+  private val processName = processDetail.substring(processDetail.indexOf('@') + 1)
+
+  private val fabricContext = FabricUtils.getFabricContext
+  private val synapseTokenServiceEndpoint: String = fabricContext(synapseTokenServiceEndpoint)
+  private val workloadEndpoint = fabricContext(TridentLakehouseTokenServiceEndpoint)
+  private val sessionToken = fabricContext(TridentSessionToken)
+  private val clusterIdentifier = fabricContext(SynapseClusterIdentifier)
+
+  def getAccessToken(resourceParam: String): String = {
+    if (!resourceMapping.contains(resourceParam)) {
+      throw new Exception(s"$resourceParam not supported")
+    }
+    val resource = resourceMapping.getOrElse(resourceParam, "")
```

Comment:
if you check that this contains the resouce you dont need to getOrElse you can just get. Also throw an IllegalArgument exception istead of exception

Reply:
Good point. I have already checked for absence of the key, don't need getOrElse.